packages/network-controller/src/NetworkController.ts

```diff
@@ -452,16 +459,30 @@ export class NetworkController extends BaseControllerV2<
       state.providerConfig.chainId = NetworksChainId[type];
       state.providerConfig.rpcUrl = undefined;
       state.providerConfig.nickname = undefined;
+      state.providerConfig.id = undefined;
     });
     this.refreshNetwork();
   }
 
+  /**
+   * Convenience method to set the current provider config to the private providerConfig class variable.
+   */
+  setCurrentAsPreviousProvider() {
```

Comment:
Is this something we want to be part of the public API? If not, should we make this private (preferably using `#` instead of `private`)?

Reply:
Yeah should be private!